pyttb/gcp/fg.py

```diff
@@ -13,42 +13,42 @@
 @overload
 def evaluate(
     model: ttb.ktensor,
-    data: ttb.tensor,
-    weights: Optional[np.ndarray],
+    data: Union[ttb.tensor, ttb.sptensor],
     function_handle: Literal[None],
     gradient_handle: function_type,
+    weights: Optional[np.ndarray],
```

Comment:
Is there a reason you re-arranged this interface? I made the ordering aligned between estimate and evaluate to keep them as similar as possible for ease of use. I didn't create a specific test for that though, so it's not strongly enforced at the moment.